packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -78,10 +78,19 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
        * cleanup properly.
        */
       this.ref.current?.addEventListener('didDismiss', (evt: any) => {
-        const wrapper = this.wrapperRef.current!;
-        this.ref.current!.append(wrapper);
-
-        this.setState({ isOpen: false });
+        const wrapper = this.wrapperRef.current;
+        const el = this.ref.current;
+
+        /**
+         * This component might be unmounted already, if the containing
+         * element was removed while the popover was still open. (For
+         * example, if an item contains an inline popover with a button
+         * that removes the item.)
+         */
+        if (wrapper && el) {
+          el.append(wrapper);
+          this.setState({ isOpen: false });
```

Comment:
This line was also moved inside the check to avoid updating state on an unmounted component (which React doesn't like).